src/components/Toast.tsx

```diff
@@ -14,9 +19,95 @@ const _colors = {
   error: colors.red.base
 };
 
+const customStyles = {
+  [ToastPosition.TOP]: {
+    style: {
+      top: 20,
+      left: "50%"
+    },
+
+    transitions: {
+      from: { transform: "translateX(-50%) translateY(-10px)" },
+      enter: { transform: "translateX(-50%) translateY(0)" },
+      leave: { transform: "translateX(-50%) translateY(-10px)" }
+    }
+  },
+
+  [ToastPosition.TOP_LEFT]: {
+    style: {
+      top: 20,
+      left: 20
+    },
+
+    transitions: {
+      from: { transform: "translateX(-10px)" },
```

Comment:
Repeating these will increase file size. Since these are string they will won't be negligible. We can't compromise on size for readability. So, extract these into separate objects and use them here.

Reply:
replacing repeating code with a dynamic function gives slightly longer code (see [here](09073e64b804ad4ba1d569a6c146dcfd3b800401))

Reply:
That link is giving 404

Reply:
Anyways the code you have written is longer. What I asked was different.

```js
const centerTransfrom = {
  from: { transform: "translateX(-50%) translateY(-10px)" },
      enter: { transform: "translateX(-50%) translateY(0)" },
      leave: { transform: "translateX(-50%) translateY(-10px)" }
}

const customStyles = {
  [ToastPosition.TOP]: {
    style: {
      top: 20,
      left: "50%"
    },
  transitions: centerTransform
}}
```

now you can use `topTransform` at two places.

Reply:
the `centerTransform` object won't be usable by `ToastPosition.BOTTOM` because the `translateY()` is negative but needs to be positive